implementation/contracts/deposit/DepositUtils.sol

```diff
@@ -373,6 +373,7 @@ library DepositUtils {
     function seizeSignerBonds(Deposit storage _d) internal returns (uint256) {
         uint256 _preCallBalance = address(this).balance;
         IBondedECDSAKeep _keep = IBondedECDSAKeep(_d.keepAddress);
+        _keep.closeKeep();
         _keep.seizeSignerBonds();
```

Comment:
calls `markAsClosed()` twice (closeKeep and in seizeSignerBonds)

this should be the other way round:
- first seizeSignerBonds (requires OnlyWhenActive and would bail if closeKeep is called first; there's nothing to seize after closekeep free'd the bonds)
- then closeKeep - this might bail on freeBond if a bond was completely seized before

is this covered by unit-tests?

Reply:
It is, but they're unit tests and it's really an integration test we need.

This was a slight miscommunication between the `seizeSignerBonds` update and this addition here; `seizeSignerBonds` is the right place (as this allows the keep to own its own closing in that scenario) and we're working on removing this `closeKeep` call.